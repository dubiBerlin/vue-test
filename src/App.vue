<template>
  <div id="app">
    <MessageList :messages="messages" />
  </div>
</template>

<script>
import MessageList from "./components/MessageList";
export default {
  data: () => ({ messages: ["Hey John", "Howdy Paco", "Bruno Wesno"] }),

  name: "App",
  components: {
    MessageList
  }
};
</script>

<style>
</style>
