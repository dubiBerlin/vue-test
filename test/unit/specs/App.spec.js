import Vue from 'vue'
import App from '@/App'

describe('App.spec.js', () => {
  let cmp, vm
  beforeEach(() => {
    cmp = Vue.extend(App) // Create a copy of the original component
    vm = new cmp({
      data: {
        // Replace data value with this fake data
        messages: ['Hey John', 'Howdy Paco', 'Bruno Wesno']
      }
    }).$mount() // Instances and mounts the component
  })
  it('equals messages to ["Hey John", "Howdy Paco", "Bruno Wesno"]', () => {
    expect(vm.messages).toEqual(['Hey John', 'Howdy Paco', 'Bruno Wesno'])
  })
  it('has the expected html structure', () => {
    expect(cmp.$el).toMatchSnapshot()
  })
})
