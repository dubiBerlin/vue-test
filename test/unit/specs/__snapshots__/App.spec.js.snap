// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.spec.js has the expected html structure 1`] = `undefined`;
